WebPhapp/backend/server.js

```diff
@@ -142,36 +142,79 @@ app.get('/api/v1/prescriptions/:patientID', (req,res) => {
 /*
 About:
 Attempts to add a prescription for a user, while also doing validation.
-Expects an object such as:
-{patientID,
-drugID,
-quantity,
-daysValid,
-refills,
-prescriberID,
-dispensorID
+Status of 200 if successful, 400 otherwise.
+Expects an object with all integer fields:
+{
+    patientID,
+    drugID,
+    quantity,
+    daysValid,
+    refills,
+    prescriberID,
+    dispensorID
 }
     Directly in terminal:
-        By both first and last name:
-            >>> curl 'http://localhost:5000/api/v1/prescriptions/add' -H 'Accept: application/json, text/plain, /*' -H 'Content-Type: application/json;charset=utf-8' --data '{"patientID":0,"drugID":0,"quantity":"","daysValid":0,"refills":0,"prescriberID":0,"dispensorID":0}
-  To be used in Axois call:
+        >>> curl 'http://localhost:5000/api/v1/prescriptions/add' -H 'Accept: application/json, text/plain, /*' -H 'Content-Type: application/json;charset=utf-8' --data '{"patientID":0,"drugID":0,"quantity":1,"daysValid":0,"refills":0,"prescriberID":0,"dispenserID":0}'
+    To be used in Axois call:
         .post("/api/v1/prescription/add",{
             patientID: 0,
             ....
         }
 */
+app.post('/api/v1/prescriptions/add',(req,res) => {    
+    const prescription = req.body;
+
+    // validate fields exist that should
+    fields = [
+        prescription.patientID,
+        prescription.drugID,
+        prescription.quantity,
+        prescription.daysValid,
+        prescription.refills,
+        prescription.prescriberID,
+        prescription.dispenserID
+    ];
+    fieldsSet = new Set(fields);
+    if(fieldsSet.has(undefined) || fieldsSet.has(null)){
```

Comment:
Can you explain how this works? I am unfamiliar with how this validates all of the inputs.

Reply:
@mdulin2 all it does is "validate fields exists that should". This means that none of the specified fields are allowed to be either undefined or null, i.e, they must all have a given value.

Reply:
Awesome; thanks for the clarification.